R/tbl-df.r

```diff
@@ -63,19 +63,21 @@ print.tbl_df <- function(x, ..., n = NULL, width = NULL, n_extra = NULL) {
   # First, subset columns
   if (!missing(j)) {
     j <- check_names_df(j, x)
-    x <- .subset(x, j)
+    result <- .subset(x, j)
+  } else {
+    result <- x
   }
 
   # Next, subset rows
   if (!missing(i)) {
-    if (length(x) == 0) {
-      nr <- length(seq_len(nr)[i])
+    if (length(result) == 0) {
+      nr <- length(attr(x, "row.names")[i])
```

Comment:
I don't see why it might be an issue. For DF with 0 columns `attr(x, "row.names")` returns the appropriate vector. Also the `result` value seems correct before the `if (length(result) == 0` command.